client/rest-high-level/src/test/java/org/elasticsearch/client/RequestTests.java

```diff
@@ -437,22 +437,24 @@ public void testDeleteIndex() {
     }
 
     public void testGetSettings() throws IOException {
-        String[] indices = randomIndicesNames(1, 5);
-        GetSettingsRequest getSettingsRequest = new GetSettingsRequest().indices(indices);
+        String emptyIndexName = "";
+        String nullIndexName = null;
+        String[] randomIndices = randomIndicesNames(1, 5);
+        String[] indicesUnderTest = java.util.stream.Stream.concat(
+            java.util.Arrays.stream(randomIndices),
+            java.util.stream.Stream.of(emptyIndexName, nullIndexName)
+        ).toArray(String[]::new);
```

Comment:
can you rather do this like we do in other similar methods, along the lines as the following snippet?

```
        String[] indices = randomBoolean() ? null : randomIndicesNames(0, 5);
        FlushRequest flushRequest;
        if (randomBoolean()) {
            flushRequest = new FlushRequest(indices);
        } else {
            flushRequest = new FlushRequest();
            flushRequest.indices(indices);
        }
```

Reply:
This one confuses me - you asked me to ensure that I tested null and empty index names, your snippet does not appear to do that.  Further, the snippet itself smells -- the if-conditional appears to simply take one of two routes to constructing an object that should be the same either way.  If we want to test our constructor logic, a unit test would accomplish that more simply

Reply:
Don't we test `null` by randomly setting indices to `null`, and empty by using `0` as a lower bound when calling `randomIndicesNames(0, 5)`? Maybe we mean to test two different things here.

I asked this because we previously had bugs in case indices were set to null, as in NPE when converting the request object.

The if is probably not so important but it just makes sure that however you set the indices (there are two ways to do it), the behaviour is the same. We are not really testing the request here or its constructor, but rather that we can convert the request object (e.g. `FlushRequest`) into our own proper internal `Request` representation. I suppose we could take the if out if we wanted to, as we effectively rely on the indices getter here. I don't think testing the two codepaths hurts though either.

Could you elaborate on what smells? I don't follow.

Reply:
Yes - I misinterpreted what you said - i thought you wanted me to test when the index name itself was null, or when the index name was "" (empty string).  I can absolutely test for when the list of index names are null or empty.  

The smell for me is in doing a /random/ test of the two ways we can set indices -- it seems to me that objective is better accomplished using a unit test that just runs all of the time and asserting that the request objects are identical.

Reply:
you are right on this, randomized tests need to be used with care, and sometimes we use them to have better coverage with a reduced amount of tests. Let's say this if addresses some paranoia of mine given that there are two ways to set indices, but I am too lazy to write two different tests for this, especially given that we are not testing the request object in this test :) It's an interesting (and potentially long) discussion, I don't mind if you want to get rid of the if.

Reply:
Should we really be test for null and empty here?  We don't do that in all of the RequestTests -- the way the rest endpoint is structured null and empty should not be possible, and we're going to ugly a few things up if we try to account for that

Reply:
This request is a bit different from the previous BroadcastRequests that we already test this for. The reason is that the serialization for GetSettingsRequest code assumes a non null value is set, and the default value is non null. We could then simply not accept null when it gets set (failing early is better). As for empty, I think it's a perfectly valid value and we should test it?

But we do test this in RequestTests for other requests.

The reason why I tend to test these corner cases is that we had issues before, as we were trying to concatenate null indices with the rest of the endpoint when building the final url. We need to make sure that we output the proper endpoint and we don't break with whatever value the request allows to set. I don't follow what the REST endpoint has to do with this, we are using the request from the client, outside of Elasticsearch.